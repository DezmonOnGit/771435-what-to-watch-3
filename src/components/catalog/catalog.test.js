import React from "react";
import renderer from "react-test-renderer";
import Catalog from "./catalog.jsx";

const filter = [`test-filter-1`, `test-filter-2`, `test-filter-3`];

const movies = [
  {
    name: `test-movie-name-1`,
    img: {
      path: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      width: 250,
      height: 125
    },
    href: `movie-page.html`
  }
];

const filterClasses = {
  list: `test-list`,
  item: `test-item`,
  itemActive: `test-item--active`,
  link: `test-link`
};

describe(`Render catalog`, () => {

  it(`Should render <Catalog />`, () => {
    const tree = renderer.create(
        <Catalog
          filter={filter}
          movies={movies}
          filterClasses={filterClasses}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
