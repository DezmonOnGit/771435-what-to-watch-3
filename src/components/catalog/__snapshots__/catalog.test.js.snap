// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render catalog Should render <Catalog /> 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <ul
      className="catalog__genres-list"
    >
      <li
        className="catalog__genres-item catalog__genres-item--active"
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          test-filter-1
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          test-filter-2
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          test-filter-3
        </a>
      </li>
    </ul>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="test-movie-name-1"
            height={125}
            src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            width={250}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
          >
            test-movie-name-1
          </a>
        </h3>
      </article>
    </div>
  </section>
</div>
`;
